src/Native/System.Security.Cryptography.Native/pal_ssl.cpp

```diff
@@ -41,6 +43,256 @@ extern "C" SSL_CTX* SslCtxCreate(SSL_METHOD* method)
     return SSL_CTX_new(method);
 }
 
+extern "C" int64_t SslCtxCtrl(SSL_CTX* ctx, int32_t cmd, int64_t larg, void* parg)
+{
+    return SSL_CTX_ctrl(ctx, cmd, larg, parg);
+}
+
+extern "C" SSL* SslCreate(SSL_CTX* ctx)
+{
+    return SSL_new(ctx);
+}
+
+extern "C" int32_t SslGetError(SSL* ssl, int32_t ret)
+{
+    return SSL_get_error(ssl, ret);
+}
+
+extern "C" void SslDestroy(SSL* ssl)
+{
+    if (ssl)
+    {
+        SSL_free(ssl);
+    }
+}
+
+extern "C" void SslCtxDestroy(SSL_CTX* ctx)
+{
+    if (ctx)
+    {
+        SSL_CTX_free(ctx);
+    }
+}
+
+extern "C" const char* SSLGetVersion(SSL* ssl)
+{
+    return SSL_get_version(ssl);
+}
+
+static CipherAlgorithmType MapCipherAlgorithmType(const SSL_CIPHER* cipher)
+{
+    unsigned long enc;
+
+#ifdef HAVE_SSL_CIPHER_ALGORITHMS
+    const unsigned long SSL_ENC_MASK = 0x1C3F8000L;
```

Comment:
I'm equally confused.  I see this for 0.9.8zg in ssl_locl.h.  If that's not automatically in the include chain, should we just include it?  At least for the 0.9.8 series?